src/components/hooks/useOnNetworkReconnect.js

```diff
@@ -0,0 +1,27 @@
+import {useRef, useContext, useEffect} from 'react';
+import {NetworkContext} from '../OnyxProvider';
+
+/**
+ * @param {Function} onNetworkReconnect
+ */
+export default function useOnNetworkReconnect(onNetworkReconnect) {
+    const callback = useRef(onNetworkReconnect);
+    callback.current = onNetworkReconnect;
```

Comment:
Yes because the function itself could change e.g. if it was passed as a `prop`. Not saying it would be the _best_ idea, but I think we rather support that than confuse someone down the line.